Common/tests/Helpers/SimplePullQueueStorage.cs

```diff
@@ -31,38 +31,38 @@ namespace ArmoniK.Core.Common.Tests.Helpers;
 
 public class SimplePullQueueStorage : IPullQueueStorage
 {
-  public readonly ConcurrentBag<string> Messages;
-
-  public SimplePullQueueStorage()
-    => Messages = new ConcurrentBag<string>();
+  public readonly ConcurrentBag<string> Messages = new();
 
   public Task<HealthCheckResult> Check(HealthCheckTag tag)
     => Task.FromResult(HealthCheckResult.Healthy());
 
   public Task Init(CancellationToken cancellationToken)
     => Task.CompletedTask;
 
-  public int MaxPriority { get; } = 10;
+  public int MaxPriority
+    => 10;
+
 
-#pragma warning disable CS1998
-  public async IAsyncEnumerable<IQueueMessageHandler> PullMessagesAsync(int nbMessages,
-#pragma warning restore CS1998
-                                                                        [EnumeratorCancellation] CancellationToken cancellationToken = default)
-  {
-    var i = 0;
-    while (i < nbMessages && Messages.TryTake(out var m))
-    {
-      i++;
-      cancellationToken.ThrowIfCancellationRequested();
-      yield return new SimpleQueueMessageHandler
-                   {
-                     CancellationToken = CancellationToken.None,
-                     TaskId            = m,
-                     MessageId = Guid.NewGuid()
-                                     .ToString(),
-                     Status            = QueueMessageStatus.Running,
-                     ReceptionDateTime = DateTime.UtcNow,
-                   };
-    }
-  }
+  public IAsyncEnumerable<IQueueMessageHandler> PullMessagesAsync(int               nbMessages,
+                                                                  CancellationToken cancellationToken = default)
+    // using ToAsyncEnumerable avoids using an async function needlessly
+    => Enumerable.Repeat(0,
```

Comment:
If you don't actually need the value, I think it might be slightly better to use a unit type instead of zero. The only I know is `ValueTuple`:
```suggestion
    => Enumerable.Repeat(new ValueTuple(),
```